flocker/node/agents/test/test_blockdevice.py

```diff
@@ -955,24 +1364,293 @@ def mountroot_for_test(test_case):
+        """
+        self.assertRaises(
+            TypeError, DestroyBlockDeviceDataset, volume=object()
+        )
+
+    def test_equal(self):
+        """
+        Two ``DestroyBlockDeviceDataset`` instances compare as equal if they
+        are initialized with the same volume.
+        """
+        dataset_id = uuid4()
+
+        def volume():
+            # Avoid using the same instance, just provide the same data.
+            return BlockDeviceVolume(
+                blockdevice_id=u"abcd",
+                size=REALISTIC_BLOCKDEVICE_SIZE,
+                dataset_id=dataset_id,
             )
+        a = DestroyBlockDeviceDataset(volume=volume())
+        b = DestroyBlockDeviceDataset(volume=volume())
+        self.assertTrue(a == b)
+
+    def test_not_equal(self):
+        """
+        Two ``DestroyBlockDeviceDataset`` instances compare as not equal if
+        they are initialized with different volumes.
+        """
+        a = DestroyBlockDeviceDataset(volume=BlockDeviceVolume(
+            blockdevice_id=u"abcd",
+            size=REALISTIC_BLOCKDEVICE_SIZE,
+            dataset_id=uuid4(),
+        ))
+        b = DestroyBlockDeviceDataset(volume=BlockDeviceVolume(
+            blockdevice_id=u"dcba",
+            size=REALISTIC_BLOCKDEVICE_SIZE,
+            dataset_id=uuid4(),
+        ))
+        self.assertTrue(a != b)
+
+    def verify_run_log(self, logger):
+        # One action is logged
+        action = assertHasAction(
+            self, logger, DESTROY_BLOCK_DEVICE_DATASET, succeeded=True)
+        all_such_actions = LoggedAction.of_type(
+            logger.messages, DESTROY_BLOCK_DEVICE_DATASET)
+        self.assertEqual([action], all_such_actions)
+        # Child actions are logged
+        [unmount] = LoggedAction.of_type(logger.messages, UNMOUNT_BLOCK_DEVICE)
+        [detach] = LoggedAction.of_type(logger.messages, DETACH_VOLUME)
+        [destroy] = LoggedAction.of_type(logger.messages, DESTROY_VOLUME)
+        self.assertEqual([unmount, detach, destroy], action.children)
+
+    @validate_logging(verify_run_log)
+    def test_run(self, logger):
+        """
+        After running ``DestroyBlockDeviceDataset``, its volume has been
+        unmounted, detached, and destroyed.
+        """
+        self.patch(blockdevice, "_logger", logger)
+
+        node = u"192.0.2.3"
+        dataset_id = uuid4()
+        api = loopbackblockdeviceapi_for_test(self)
+        volume = api.create_volume(
+            dataset_id=dataset_id, size=REALISTIC_BLOCKDEVICE_SIZE
+        )
+        volume = api.attach_volume(volume.blockdevice_id, node)
+        device = api.get_device_path(volume.blockdevice_id)
+        mountroot = mountroot_for_test(self)
+        mountpoint = mountroot.child(unicode(dataset_id).encode("ascii"))
+        mountpoint.makedirs()
+        check_output([b"mkfs", b"-t", b"ext4", device.path])
+        check_output([b"mount", device.path, mountpoint.path])
```

Comment:
There's no generic mount code you can use yet? I guess it's part of combo statechange for now
.